Stop specifying netgo on static builds

mberhault opened the pull request:
I don't think we need it anymore, and the suffix is making teamcity builds unhappy.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/examples-go/83)
<!-- Reviewable:end -->

tschottdorf commented:
Cc @tamird

tamird commented:
Since static builds only happen on Linux, I think this is OK? You checked in some random jemalloc stuff, though.

Also, the commit message needs more words: what's up with TC builds?

mberhault commented:
oops, removed the extra heap profiles.

The teamcity build never tried to build the binaries, it just ran the tests.
I added the build rule to build/push the binaries to AWS.

tamird commented:
Oh, derp, I thought this was the main repo. LGTM
